pkg/dashboard/resource/function.go

```diff
@@ -112,20 +112,16 @@ func (fr *functionResource) Create(request *http.Request) (id string, attributes
 		return
 	}
 
+	// TODO: Add a lock to prevent race conditions here (prevent 2 functions created with the same name)
 	// validate there are no 2 functions with the same name
-	getFunctionsOptions := &platform.GetFunctionsOptions{
+	functions, err := fr.getPlatform().GetFunctions(&platform.GetFunctionsOptions{
 		Name:      functionInfo.Meta.Name,
 		Namespace: fr.getNamespaceFromRequest(request),
-	}
-
-	// TODO: Add a lock to prevent race conditions here (prevent 2 functions created with the same name)
-	functions, err := fr.getPlatform().GetFunctions(getFunctionsOptions)
+	})
 	if err != nil {
 		responseErr = nuclio.WrapErrInternalServerError(errors.Wrap(err, "Failed to get functions"))
 		return
-	}
-
-	if len(functions) > 0 {
+	} else if len(functions) > 0 {
```

Comment:
```suggestion
	}
	if len(functions) > 0 {
```

No need for else after return

Reply:
Thanks for the comment! I did it to just keep this logic in one block, so it'll be easier to read later